tensorflow/compiler/xla/service/all_reduce_reassociate.cc

```diff
@@ -77,17 +166,55 @@ StatusOr<bool> AllReduceReassociate::Run(
       }
 
       // Found pattern op(ar(x), ar(y)). Transform it into ar(op(x,y)).
-      HloInstruction *new_op = computation->AddInstruction(
-          inst->CloneWithNewOperands(inst->shape(), {ar0->mutable_operand(0),
-                                                     ar1->mutable_operand(0)}));
+      HloInstruction *new_op_operand0 = ar0->mutable_operand(0);
+      HloInstruction *new_op_operand1 = ar1->mutable_operand(0);
+      if (convert0) {
+        HloInstruction *ar0_operand = ar0->mutable_operand(0);
+        TF_RETURN_IF_ERROR(convert0->ReplaceOperandWith(0, ar0_operand));
```

Comment:
This is ok only if the convert does not have any other use. May be add that to `MatchOperandsToAllReduceWithOptionalConvert`?